.editorconfig

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Configuration here must match the one for Spotless in `pom.xml`!
+
+root = true
+
+[*]
+indent_size = 4
+ij_continuation_indent_size = 8
```

Comment:
While this matches the Kotlin standard code style, I initially created this project with 2 and 4 as I much prefer it over 4 and 8. Either we should change this to 2 and 4, or do a project-wide change to 4 and 8. I'd prefer the former.

Reply:
The 2 and 4 style also matches the [`google-java-format`](https://github.com/google/google-java-format) style, which has a great advantage: it is deterministic.

Right now on the main repo does not enforce a deterministic style (e.g. continuation indentation is fine as long as it does not have tabs). We might want to switch to a deterministic one and Google Java Format is a valid contender.

Reply:
Related to this and to the spotbugs reports: in the Kotlin world, `detekt` is a really good static code analyzer. It works on the AST rather than the bytecode, but it has rules for both formatting and spots common coding issues similar to Spotbugs.

I could introduce detekt to this project in a future PR.

Reply:
I have updated it to 2/4 in 9100bc950b4e38522a3a11fcb565f12163589bcf. Though I need to admit this slightly contradicts with the effort of unifying the styling across Log4j projects. I would prefer this to be part of Spotless configuration in `logging-parent`. `.editorconfig` and Spotless should move in tandem.

Regarding `detekt`...  We can add it individually. But again, I prefer [`detekt` being added to Spotless](/diffplug/spotless/issues/143) instead.